### ISSUE TITLE
Automatic key detection for JsonProperty at backed enum cases

### ISSUE THREAD

**Issue by JuraSciix** [JuraSciix/Objeckson#1]
Condition:
When programmer doesn't specify the key explicitly, but does specify a backed enum value, use the value as the key.

Example:
```php
<?php

use jurasciix\objeckson\JsonProperty;

enum TaskStatus: string {
    #[JsonProperty]
    case PENDING = 'new'; // automatically detects the key as 'new'

    #[JsonProperty('closed')] // uses the key from attribute
    case CLOSED = 'closed';
}
```

**Comment by JuraSciix** [comment]
If neither the key nor the backed value is specified, then use the case name (with the case).

**Comment by JuraSciix** [comment]
Implemented in c8f47a78ed80f079a7786104e6ad20cf53cae121
